server/src/main/java/org/opensearch/search/pipeline/PipelinedRequestContext.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.search.pipeline;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * A holder for state that is passed through each processor in the pipeline.
+ */
+public class PipelinedRequestContext {
+    private final Map<String, Object> genericRequestContext = new HashMap<>();
+
+    public Map<String, Object> getGenericRequestContext() {
```

Comment:
Should we hide the details that this is `Map`? May be have `put` / `get` / `getAll` APIs instead?

Reply:
Maybe. 

I think I like the idea of having just `put` and `get`, since I would prefer that the API express the opinion that you should know what variables you're reading or writing.

Originally, the `Map` **was** the context object, but @navneet1v suggested wrapping it in a class for future extensibility in the form of added fields. For example, I'm considering the idea of optionally attaching the `IdentityService.getSubject()` to the context (resolved on the original REST action thread) so that it can be used by processors. (That would be a final, read-only context field.)

Anyway, I think I like the idea of encapsulating the map behind `void putGenericVariable(String, Object)` and `Object getGenericVariable(String)` methods. (Actually, since I was inspired by Java servlets, I could go with `putAttribute` and `getAttribute`.)

Reply:
I went with `getAttribute/setAttribute`. 

@reta, what do you think?

Reply:
> @reta, what do you think?

Love it, thanks @msfroh !